lib/workers/repository/update/pr/index.ts

```diff
@@ -154,15 +155,9 @@ export type EnsurePrResult = ResultWithPr | ResultWithoutPr;
 export async function ensurePr(
   prConfig: BranchConfig
 ): Promise<EnsurePrResult> {
-  let branchStatus: BranchStatus;
-  async function getBranchStatus(): Promise<BranchStatus> {
-    if (branchStatus) {
-      return branchStatus;
-    }
-    branchStatus = await resolveBranchStatus(branchName, ignoreTests);
-    logger.debug(`Branch status is: ${branchStatus}`);
-    return branchStatus;
-  }
+  const getBranchStatus = memoize<Promise<BranchStatus>>(() =>
```

Comment:
```suggestion
  const getBranchStatus = memoize(() =>
```
shouldn't it work without explicit type arg?

Reply:
Cool, didn't expect type inference would handle it